File: backend/health-data-service/src/main/java/com/health/data/dto/DailyMetricsDto.java
package com.health.data.dto;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DailyMetricsDto {
    private String userId;
    private int steps;
    private Long sleepStartTime; // nullable
    private Long sleepEndTime;   // nullable
    private Long sleepDuration;  // nullable
    private String date;         // format: yyyy-MM-dd
}

File: app/src/main/java/com/health/healthmonitorwearapp/network/DailyMetricsDto.kt
package com.health.healthmonitorwearapp.network

data class DailyMetricsDto(
    val userId: String,
    val steps: Int,
    val sleepStartTime: Long?,   // optional but can be useful
    val sleepEndTime: Long?,     // optional
    val sleepDuration: Double?,    // primary
    val date: String
)
